src/coreComponents/constitutive/solid/SolidExtrinsicData.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SolidExtrinsicData.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_SOLID_SOLIDEXTRINSICDATA_HPP
+#define GEOSX_CONSTITUTIVE_SOLID_SOLIDEXTRINSICDATA_HPP
+
+#include "mesh/ExtrinsicMeshData.hpp"
+
+namespace geosx
+{
+
+namespace extrinsicMeshData
+{
+
+namespace solid
+{
+
+EXTRINSIC_MESH_DATA_TRAIT( bulkModulus,
```

Comment:
This is a subtle issue, but this approach implicitly assumes we have an elastic model with a constant bulk modulus and shear modulus as fixed parameters.  Many of our material models don't directly use these parameters (e.g. anisotropic elasticity,  pressure-dependent elasticity, and modified Cam-Clay) and their value actually depends on the loading state (e.g. for a plastic model the bulk modulus may soften with time from an initial value).  I would instead suggest adding a method in the material models themselves to return the current value, e.g. `solidModel.getBulkModulus(k,q)` which we can use in each cell to construct an appropriate scaling constant (probably lagged in time to avoid nonlinearities).  The thing to register on the mesh is probably the scaling constant itself, which can be updated at the end of each timestep in a loop. I recognize this may be overkill, but the current implementation may not work for a large class of our material models.  Perhaps we can discuss this more in person.

Reply:
This was a great point! I have updated things to reflect this suggested structure.